fork-management-for-infra-providers/README.md

```diff
@@ -0,0 +1,148 @@
+  - https://github.com/giantswarm/cluster-api-provider-cloud-director-app
+  - https://github.com/giantswarm/cluster-api-provider-gcp-app
+
+### Syncing fork repositories
+
+There is no common way.
+
+In CAPVCD, since we have some diffs between our fork and upstream repo, we have a mechanism as follows:
+- `main` branch of the repo is the same as the upstream one. We sync this branch by clicking `Sync fork` button in the UI.
+- `giantswarm-main` branch of the repo contains our changes. We manually rebase it on top of `main` branch when necessary and open a new PR each time. The developer is supposed to resolve conflicts during rebasing.
+ 
+### Bundling + Customization
+
+There are some differences between repositories but the general structure in app repositories is as follows:
+
+- We have Makefile + Bash scripts to
+  - Check out a specific commit/branch of fork/upstream repo. [Example](https://github.com/giantswarm/cluster-api-provider-cloud-director-app/blob/adfc74115d9b7aeda60d885500122be3fb434d2e/Makefile.custom.mk#L14)
+  - Generate base manifests by running scripts of the core repo
+  - Customize base manifests with kustomization. [Example](https://github.com/giantswarm/cluster-api-provider-cloud-director-app/blob/adfc74115d9b7aeda60d885500122be3fb434d2e/Makefile.custom.mk#L19)
+  - Update manifests under `helm` folder for the app. [Example](https://github.com/giantswarm/cluster-api-provider-cloud-director-app/blob/adfc74115d9b7aeda60d885500122be3fb434d2e/Makefile.custom.mk#L28)
+
+When it is necessary to update the app content, developers are supposed to
+- run make targets manually to update manifests and values in the app repo
+- open a PR in the app repo
+
+### Image Management
+
+For some providers, we retag upstream images with retagger.
+For some providers like CAPVCD, we build/push images manually.
+
+## Design proposals
+
+### 1. Repo management
+
+#### 1.a Separate repositories for fork and app
+
+We can have two repositories per operator. One is only fork of upstream and the other one is for only bundling it as an app.
+
+#### 1.b One repository for fork and app
+
+We can maintain bundling and customization logic in the same repo.
+
+#### Decision
+
+- `1.a` is the current situation.
+- `1.b` will lead to many conflicts with the upstream repository and it will be hard to automate conflict resolution.
+
+`1.a` is decided.
+
+---
+
+### 2. Branching strategy in fork repo
+
+#### 2.a main vs giantswarm-main
+
+We can maintain two branches in the fork repo. 
+- `main` will be just the same as `main` of upstream repository. This branch can be synced automatically.
+- `giantswarm-main` will contain our changes if necessary. 
+
+The update flow will be
+- First update `main`
+- Then rebase `giantswarm-main`
+
+#### 2.b only main
+
+We can maintain only one `main` branch including our changes and use `git merge` to apply upstream changes.
+
+#### Decision
+
+`2.a` is decided.
```

Comment:
Who decided this? 

This wont work for CAPA, for example, as the `main` branch isn't a release branch. For CAPA we're based on the `release-1.5` branch until we have the bandwidth to upgrade to the next `release-xxx` branch but we would never release from a `main` branch.

Reply:
> but we would never release from a main branch

We should never release from `main` branch in any provider, at least not according to upstream branching/versioning strategy where releases are coming only from `release-xxx` branches.